### PR TITLE
Kernel/Net: Enable usage of RTL8168E and RTL8168H in RTL8168 driver

### PR DESCRIPTION
**Pull request by supercomputer7**
cc @IdanHo

### CODE REVIEW

File: Kernel/Net/RTL8168NetworkAdapter.cpp [hunk 1]
```diff
@@ -389,8 +391,10 @@ void RTL8168NetworkAdapter::configure_phy()
         TODO();
     case ChipVersion::Version14:
         TODO();
-    case ChipVersion::Version15:
-        TODO();
+    case ChipVersion::Version15: {
+        configure_phy_e_2();
```
Comment (IdanHo) [inline Kernel/Net/RTL8168NetworkAdapter.cpp:395]: It's pretty weird that this works, linux's configuration for this is pretty different, not much we can do without a specification though :man_shrugging:

**Comment by IdanHo** [comment]
BTW, can you add the working variants to https://github.com/SerenityOS/serenity/blob/master/Documentation/HardwareCompatibility.md#network-adapters as well?

**Comment by supercomputer7** [comment]
> BTW, can you add the working variants to https://github.com/SerenityOS/serenity/blob/master/Documentation/HardwareCompatibility.md#network-adapters as well?

Sure thing.

File: Kernel/Net/RTL8168NetworkAdapter.cpp (hunk 1 above)
Comment (supercomputer7) [reply Kernel/Net/RTL8168NetworkAdapter.cpp:395]: I know, but it works, so... I guess we can look into the linux driver to see exactly what's going on there. I will leave it as a task for another PR :)

File: Kernel/Net/RTL8168NetworkAdapter.cpp [hunk 2]
```diff
@@ -207,8 +207,10 @@ UNMAP_AFTER_INIT RTL8168NetworkAdapter::RTL8168NetworkAdapter(PCI::Address addre
     identify_chip_version();
     dmesgln("RTL8168: Version detected - {} ({}{})", possible_device_name(), (u8)m_version, m_version_uncertain ? "?" : "");
     if (m_version == ChipVersion::Unknown || (m_version >= ChipVersion::Version4 && m_version <= ChipVersion::Version16) || m_version >= ChipVersion::Version18) {
-        dmesgln("RTL8168: Aborting initialization! Support for your chip version ({}) is not implemented yet, please open a GH issue and include this message.", (u8)m_version);
-        return; // Each ChipVersion requires a specific implementation of configure_phy and hardware_quirks
+        if (m_version != ChipVersion::Version15 && m_version != ChipVersion::Version30) {
```
[the remark this answers is not in the document]
Comment (supercomputer7) [reply Kernel/Net/RTL8168NetworkAdapter.cpp]: I wrote a method with a switch case statements to solve this :)

File: Kernel/Net/RTL8168NetworkAdapter.cpp [hunk 3]
```diff
@@ -192,6 +192,51 @@ UNMAP_AFTER_INIT RefPtr<RTL8168NetworkAdapter> RTL8168NetworkAdapter::try_to_ini
...
+    case ChipVersion::Version25:
+    case ChipVersion::Version26:
+    case ChipVersion::Version27:
+    case ChipVersion::Version28:
+    case ChipVersion::Version29:
+        return false;
+    case ChipVersion::Version30:
+        return true;
+    default:
+        VERIFY_NOT_REACHED();
```
Comment (IdanHo) [inline Kernel/Net/RTL8168NetworkAdapter.cpp:236]: I dont think we should crash for an unknown version (ChipVersion::Unknown), we should just return false as well.

Comment (supercomputer7) [reply Kernel/Net/RTL8168NetworkAdapter.cpp:236]: Right, I didn't take that into consideration.

**Review by IdanHo — COMMENTED** [review]
Last review comment, I swear :^)
1st commit: `or let drivers to utilize other ranges` => `or let drivers utilize other ranges`
2nd commit: `and they seem to work just fine with driver` => `and they seem to work just fine with the driver`

**Comment by supercomputer7** [comment]
> Last review comment, I swear :^)
> 1st commit: `or let drivers to utilize other ranges` => `or let drivers utilize other ranges`
> 2nd commit: `and they seem to work just fine with driver` => `and they seem to work just fine with the driver`

These are my stupid grammar mistake, please always let me know about them so I can fix them (and get them fixed in future contributions) :)

**Review by IdanHo — APPROVED** [review]
LGTM, I'm going to wait for CI just in case (due to the first commit)

**Pull request merged**